src/components/ApplicationDetails/ApplicationHeader.tsx

```diff
@@ -54,25 +43,30 @@ export const ApplicationHeader: React.FC<{ application: ApplicationKind }> = ({
           ) : null}
         </Flex>
         <FlexItem>
-          {selectedComponentRoute && (
-            <>
-              <ExternalLink href={selectedComponentRoute}>
-                {selectedComponentRoute.length > 40
-                  ? `${selectedComponentRoute.slice(0, 40)}...`
-                  : selectedComponentRoute}
-              </ExternalLink>{' '}
-              <Tooltip content={urlCopied ? 'URL Copied' : 'Copy URL'}>
-                <Button
-                  variant="plain"
-                  data-testid="route-copy-icon"
-                  isInline
-                  onClick={onCopyButtonClick}
-                >
-                  <OutlinedCopyIcon color={linkColor.value} />
-                </Button>
+          {selectedComponentRoute &&
+            (selectedComponentRoute.length > 40 ? (
```

Comment:
`Truncate` will only truncate if necessary. Therefore I think we can always render this path without the conditional rendering.